tutorial/tests/testsuite_helpers.py

```diff
@@ -257,24 +323,45 @@ class ResultCollector:
     def __init__(self) -> None:
         self.tests: Dict[str, TestResult] = {}
 
+    def pytest_runtest_makereport(self, item: pytest.Item, call: pytest.CallInfo):
```

Comment:
As I said above, we should also investigate the case where `call.when` is during test collection but the collection fails for a syntax error.